src/WebJobs.Script.WebHost/App_Start/WebHostSettings.cs

```diff
@@ -1,13 +1,29 @@
 ﻿// Copyright (c) .NET Foundation. All rights reserved.
 // Licensed under the MIT License. See License.txt in the project root for license information.
 
+using Microsoft.Azure.WebJobs.Host;
+using Newtonsoft.Json;
+
 namespace Microsoft.Azure.WebJobs.Script.WebHost
 {
     public class WebHostSettings
     {
+        [JsonProperty("isSelfHost")]
         public bool IsSelfHost { get; set; }
+
+        [JsonProperty("scriptPath")]
         public string ScriptPath { get; set; }
+
+        [JsonProperty("logPath")]
         public string LogPath { get; set; }
+
+        [JsonProperty("secretsPath")]
         public string SecretsPath { get; set; }
+
+        [JsonProperty("nodeDebugPort")]
+        public int NodeDebugPort { get; set; }
+
+        [JsonIgnore]
+        public TraceWriter TraceWriter { get; set; }
```

Comment:
Also, I'm thinking that for the purposes of returning a DTO from the admin api, it might be good to just define a new type that defines what you want to return. This type is kind of an internal settings class that may grow over time, and I don't want that to result in us returning more than we intended inadvertently.